mp/src/game/server/player.cpp

```diff
@@ -2187,22 +2187,7 @@ void CBasePlayer::PlayerDeathThink(void)
 	{
 		fAnyButtonDown &= ~IN_DUCK;
 	}
-
-#ifndef NEO
-	// wait for all buttons released
-	if (m_lifeState == LIFE_DEAD)
-	{
-		if (fAnyButtonDown)
-			return;
-
-		if ( g_pGameRules->FPlayerCanRespawn( this ) )
-		{
-			m_lifeState = LIFE_RESPAWNABLE;
-		}
-
-		return;
-	}
-#endif
```

Comment:
We generally should not touch non-NT engine code destructively, as it can help us with acting as reference to the engine baseline behaviour. So, I would prefer we not delete this code but rather keep the ifndef guard to disable it.

Or, as it seems we'd never want to hit this base implementation code, we could instead remove the redundant `#ifndef NEO` guard, and replace it with something like:
```cpp
void CBasePlayer::PlayerDeathThink(void)
{
#ifdef NEO
	Assert(false); // we should never reach this
#endif
```
at the top of the base implementation to catch it as a smoke test.

...or just remove the ifdef guard and leave it at that, if we're confident this won't be called into.